server/src/test/java/org/apache/calcite/avatica/server/BasicAuthHttpServerTest.java

```diff
@@ -44,8 +44,8 @@
   private static String url;
 
   @BeforeClass public static void startServer() throws Exception {
-    final String userPropertiesFile = URLDecoder.decode(BasicAuthHttpServerTest.class
-        .getResource("/auth-users.properties").getFile(), "UTF-8");
+    final String userPropertiesFile = (new File(BasicAuthHttpServerTest.class
+        .getResource("/auth-users.properties").toURI())).getPath();
```

Comment:
```suggestion
    final String userPropertiesFile = Sources.of(BasicAuthHttpServerTest.class.getResource("/auth-users.properties"))
        .file().getAbsolutePath();
```

Please use `org.apache.calcite.util.Sources`

Reply:
@vlsi 
AFAIK Avatica doesn't depend on Calcite by design, and org.apache.calcite.util.Sources is not part of Avatica.

Would you like me to copy that class into Avatica, and use it ?

Reply:
Oh, I missed it is `avatica`.
I think it makes sense to copy the class or something like that.

I do remember there is a lot of edge cases near File vs URL vs URI, so I would like to have a single class for conversion between URL and Files.

Reply:
I have copied most of Sources.java and Source.java to Avatica.
I've put them in core, but it may also make sense to put them in server or even server/tests.